src/graph/graph.c

```diff
@@ -224,28 +170,15 @@ Graph *Graph_New(size_t n) {
     assert(n > 0);
     Graph *g = malloc(sizeof(Graph));
     
-    g->node_cap = GRAPH_NODE_COUNT_TO_BLOCK_COUNT(n) * NODEBLOCK_CAP;
-    g->node_count = 0;
+    g->nodes = DataBlock_New(n, sizeof(Node));
+    g->edges = DataBlock_New(n, sizeof(Edge));
     g->relation_cap = GRAPH_DEFAULT_RELATION_CAP;
     g->relation_count = 0;
     g->label_cap = GRAPH_DEFAULT_LABEL_CAP;
     g->label_count = 0;
-    
-    g->block_count = GRAPH_NODE_COUNT_TO_BLOCK_COUNT(n);
-    g->nodes_blocks = malloc(sizeof(NodeBlock*) * g->block_count);
-    
-    // Allocates blocks.
-    for(int i = 0; i < g->block_count; i++) {
-        g->nodes_blocks[i] = NodeBlock_New();
-        if(i > 0) {
-            // Link blocks.
-            g->nodes_blocks[i-1]->next = g->nodes_blocks[i];
-        }
-    }
-
     g->_relations = malloc(sizeof(GrB_Matrix) * g->relation_cap);
     g->_labels = malloc(sizeof(GrB_Matrix) * g->label_cap);
-    GrB_Matrix_new(&g->adjacency_matrix, GrB_BOOL, g->node_cap, g->node_cap);
+    GrB_Matrix_new(&g->adjacency_matrix, GrB_BOOL, _Graph_NodeCap(g), _Graph_NodeCap(g));
```

Comment:
`_Graph_Resize_Matrix` (immediately invoked in `Graph_CreateNodes`) will shrink matrices to NodeCount before they get populated, so that would probably be a better allocation size.

Reply:
No, that can only be 0 at this point. We can arrive here from `_MGraph_Query` or `_MGraph_BulkInsert` (through `_MGraph_CreateGraph`); in either case I think we can make an upper-bound estimate pretty well, and that can be passed as an argument to this function.

For a query, this upper bound could be `Vector_Size(ast->createNode->graphEntities)`. which would actually include the edge count as well, but I expect would still generally be closer.

For a bulk insert, the total node count is one of the first tokens in `argv`, which we can access directly or defer graph creation slightly until it has been read in order.
 
`Graph_CreateNodes` will immediately resize this 16,384-column matrix down to node count, and every label matrix will be similarly resized upon retrieval, so I do think a more informed starting size would be preferable.